public/app/features/integrated-alerting/components/AlertRuleTemplate/AddAlertRuleTemplateModal/AddAlertRuleTemplateModal.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FC, ChangeEvent, useCallback, useRef } from 'react';
+import { FormApi } from 'final-form';
+import { Form } from 'react-final-form';
+import { Button, HorizontalGroup, Icon, useStyles } from '@grafana/ui';
+import { Modal, LoaderButton, TextareaInputField, validators } from '@percona/platform-core';
+import { Messages } from 'app/features/integrated-alerting/IntegratedAlerting.messages';
+import { AddAlertRuleTemplateModalProps, AlertRuleTemplateRenderProps } from './AddAlertRuleTemplateModal.types';
+import { getStyles } from './AddAlertRuleTemplateModal.styles';
+import { AlertRuleTemplateService } from '../AlertRuleTemplate.service';
+
+export const AddAlertRuleTemplateModal: FC<AddAlertRuleTemplateModalProps> = ({ isVisible, setVisible }) => {
+  const styles = useStyles(getStyles);
+  const { required } = validators;
+  const inputRef = useRef<HTMLInputElement>(null);
+  const onUploadFile = useCallback(
+    (change: FormApi['change']) => (event: ChangeEvent<HTMLInputElement>) => {
+      const file = event.target.files.length > 0 ? event.target.files[0] : null;
+      const reader = new FileReader();
+
+      if (file) {
+        reader.addEventListener('load', e => change('yaml', e.target.result));
+        reader.readAsText(event.target.files[0]);
+      }
+    },
+    []
+  );
+  const onSubmit = async (values: AlertRuleTemplateRenderProps) => {
+    try {
+      await AlertRuleTemplateService.upload(values);
+      setVisible(false);
+    } catch (e) {
+      console.error(e);
```

Comment:
Please use the `logger` utility instead of console :)

Reply:
Done!